1-Draft/RFCNNNN-Implicit-Line-Continuance.md

```diff
@@ -0,0 +1,148 @@
+
+1. For either example, insert a blank line between `Get-Process` and the next command that starts with dash, as shown here:
+
+    ```PowerShell
+    function -dash {
+        'run'
+    }
+
+    Get-Process -Id $PID
+
+    -dash # Runs the -dash command because implicit line continuance stops looking when it encounters a blank line
+
+    Get-Process -Id $PID
+
+    -split 'a b c d' # splits the string 'a b c d' into an array with four items
+    ```
+
+1. In the example with a command that starts with a dash, invoke the command using the call operator, as shown here:
+
+    ```PowerShell
+    function -dash {
+        'run'
+    }
+
+    Get-Process -Id $PID
+    & -dash # Runs the `-dash` command because implicit line continuance sees the call operator and recognizes that the line does not continue further
+    ```
+
+    Unfortunately this workaround does not work for the -split or -join unary operators, because you cannot invoke statement that starts with a unary operator with the call operator (the call operator is only for invoking commands).
+
+In practice, I believe that there are not many commands out there that start with a dash. While researching this, I couldn't find a single command on Windows or Linux that starts with a dash, so that's not a very risky scenario. However, since the `-split` and `-join` unary operators exist in PowerShell, there is a good likelihood that those may be used on a line following a command, and that is where this breaking change would have the most potential impact. While I don't feel that risk is enough to reject this proposal entirely, because it adds significant value to script authors, it is worth considering what the best approach would be.
+
+Special casing the named unary operators could work (among the thousands of commands on my system, none of them have split or join parameters, but some may exist somewhere), but that would mean future unary operators be special cased as well, so that approach isn't desirable because it risks future breaking changes.
+
+If the risk is deemed to be too high because of the risk with named unary operators, at a bare minimum I feel this feature offers enough significant value to the community that it should not be rejected, but instead offered as an optional feature so that users wanting the benefit can opt-into the functionality, and mark it as enabled for their scripts/modules. I also suspect the majority of scripters would want it turned on and would then simply write their scripts accordingly. It's really too bad though that unary operators with string names use the same single first character as parameter names because they get in the way here. In hindsight, named operators should probably have been prefixed with something other than a single dash to differentiate them from named parameters (something to consider if PowerShell ever comes out with a version with significant breaking changes plus conversion scripts).
```

Comment:
Would it reduce the risk if we require some character on the line after the command to initiate support for parameter line wrapping? For instance, what if we put the splat character (and ignored whitespace) at the end of the line as an indicator that the parameters continue on multiple lines:

```powershell
Get-Process @
-Id $pid
```
vs 

```PowerShell
Get-Process -Id $PID
-split 'a b c d'
```

Reply:
That's interesting. That would eliminate the breaking change factor entirely, making it opt-in. A little unfortunate to have to add something to each command you want to wrap, vs having a feature/flag that allows script/module authors to enable this for their content without impacting end users ad-hoc use.

Reply:
My initial read of this RFC was negative simply because it would change the functionality of non-visable characters in a non-obvious way. Choosing the begin wrapping with the '@' character makes the most sense to me. This is way better than using a back tick and if fully supported, should also provide intellisense. That would make this method preferable when compared to splatting which has no intellisense.

Reply:
Now I'm thinking it might be interesting if the Windows terminal supported a multi-line command mode, allowing users to enter multiple-line commands without executing them until enter is pressed after a blank line at the end of the command. And the terminal is open source now. Hmmm.....

Reply:
@ngetchell It would only be non-obvious if you followed a command with the `-splat` or `-join` unary operators (or a command whose name starts with dash, but I've yet to see one of those). Plus scripters should apply appropriate formatting to this, just like they would for wrapped pipelines, making it pretty obvious to the reader. What I like about the extra character at the end (and the other alternative I added, which happens to be my preference) is that it makes this become a non-breaking change.

Reply:
Is it worth pointing out that `Write-Host @` is currently _not_ valid with a line-ending after it, whereas, for instance `Write-Host ...` is valid currently, because `...` is interpreted as a string...